Managers/WindowMan.cpp

```diff
@@ -0,0 +1,602 @@
+			[](auto left, auto right) {
+				return left.second.x < right.second.x;
+			}
+		);
+
+		for (const auto &[displayIndex, displayBounds] : m_ValidDisplayIndicesAndBoundsForMultiDisplayFullscreen) {
+			// Translate display offsets to backbuffer offsets, where the top left corner is (0,0) to figure out if the display arrangement is unreasonable garbage, i.e not top or bottom edge aligned.
+			// If any of the translated offsets ends up negative, or over-positive for the Y offset, disallow going into multi-display fullscreen
+			// because we'll just end up with an access violation when trying to read from the backbuffer pixel array during rendering.
+			// In odd display size arrangements this is valid as long as the misaligned display is somewhere between the top and bottom edge of the tallest display, as the translated offset will remain in bounds.
+			int translatedOffsetX = (displayBounds.x - leftMostOffset);
+			int translatedOffsetY = (displayBounds.y - topMostOffset);
+			if (translatedOffsetX < 0 || translatedOffsetY < 0 || translatedOffsetY + displayBounds.h > maxHeight) {
+				setSingleDisplayMode("Bad display alignment detected!\nMulti-display fullscreen currently supports only horizontal arrangements where all displays are either top or bottom edge aligned!");
+				return;
+			}
+		}
+
+		for (const auto &[displayIndex, displayBounds] : m_ValidDisplayIndicesAndBoundsForMultiDisplayFullscreen) {
+#if	SDL_VERSION_ATLEAST(2, 24, 0)
+			m_DisplayArrangmentLeftMostDisplayIndex = SDL_GetRectDisplayIndex(&displayBounds);
+			if (m_DisplayArrangmentLeftMostDisplayIndex >= 0) {
+#else
+			// This doesn't return the nearest display index to the point but should still be reliable enough for reasonable display arrangements.
+			SDL_Point testPoint = { leftMostOffset + 1, topMostOffset + 1 };
+			if (SDL_PointInRect(&testPoint, &displayBounds) == SDL_TRUE) {
+#endif
+				m_DisplayArrangmentLeftMostDisplayIndex = displayIndex;
+				break;
+			}
+		}
+
+		if (m_DisplayArrangmentLeftMostDisplayIndex >= 0) {
+			m_MaxResX = totalWidth;
+			m_MaxResY = maxHeight;
+			m_DisplayArrangementLeftMostOffset = leftMostOffset;
+			m_DisplayArrangementTopMostOffset = topMostOffset;
+			m_CanMultiDisplayFullscreen = true;
+		} else {
+			setSingleDisplayMode("Unable to determine left-most display index!");
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void WindowMan::ValidateResolution(int &resX, int &resY, int &resMultiplier) const {
+		if (resX * resMultiplier > m_MaxResX || resY * resMultiplier > m_MaxResY || resMultiplier < 1 || resMultiplier > 8) {
+			resMultiplier = std::clamp(resMultiplier, 1, 8);
+			resX = std::min(resX, m_MaxResX / resMultiplier);
+			resY = std::min(resY, m_MaxResY / resMultiplier);
+			ShowMessageBox("Resolution too high to fit display, overriding to fit!");
+			g_SettingsMan.SetSettingsNeedOverwrite();
```

Comment:
Why does this do `g_SettingsMan.SetSettingsNeedOverwrite()` instead of `g_SettingsMan.UpdateSettingsFile()`?
Is it cause this is called from somewhere that'll then update the file if the need overwrite flag is true?

Reply:
this gets called during Init, so updating settings here is likely to cause issues